transports/webrtc/src/tokio/req_res_chan.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use futures::{
+    channel::{mpsc, oneshot},
+    SinkExt,
+};
+use futures_lite::StreamExt;
```

Comment:
```suggestion
use futures::StreamExt;
```
Why not use `futures` here?

Reply:
because `futures::StreamExt` does not provide `poll_next`?

Reply:
> because `futures::StreamExt` does not provide `poll_next`?

It provides `poll_next_unpin` which does the same thing I think.

Reply:
isn't that for types which can be unpinned? okay, I will change

Reply:
Yes. Our type is not even pinned but just requires `&mut self` :)